content/en/blog/_posts/2022-08-03-kms-v2-alpha.md

```diff
@@ -0,0 +1,89 @@
+    else not using key hierarchy
+        %% current behavior
+        kmsplugin->>externalkms: encrypt DEK with remote KEK
+        externalkms->>kmsplugin: encrypted DEK
+        kmsplugin->>kubeapiserver: return encrypt response <br/> {"ciphertext": "<encrypted DEK>", key_id: "<remote KEK ID>", "annotations": {}}
+    end
+    kubeapiserver->>etcd: store encrypt response and encrypted DEK
+```
+#### Decrypt Request
+
+```mermaid
+sequenceDiagram
+    participant kubeapiserver
+    participant kmsplugin
+    participant externalkms
+    %% if local KEK in annotations, then using hierarchy
+    alt encrypted local KEK is in annotations
+      kubeapiserver->>kmsplugin: decrypt request <br/> {"ciphertext": "<encrypted DEK>", key_id: "<key_id gotten as part of EncryptResponse>", <br/> "annotations": {"kms.kubernetes.io/local-kek": "<encrypted local KEK>"}}
+        alt encrypted local KEK in cache
+            kmsplugin->>kmsplugin: decrypt DEK with local KEK
+        else encrypted local KEK not in cache
+            kmsplugin->>externalkms: decrypt local KEK with remote KEK
+            externalkms->>kmsplugin: decrypted local KEK
+            kmsplugin->>kmsplugin: decrypt DEK with local KEK
+            kmsplugin->>kmsplugin: cache decrypted local KEK
+        end
+        kmsplugin->>kubeapiserver: return decrypt response <br/> {"plaintext": "<decrypted DEK>", key_id: "<remote KEK ID>", <br/> "annotations": {"kms.kubernetes.io/local-kek": "<encrypted local KEK>"}}
+    else encrypted local KEK is not in annotations
+        kubeapiserver->>kmsplugin: decrypt request <br/> {"ciphertext": "<encrypted DEK>", key_id: "<key_id gotten as part of EncryptResponse>", <br/> "annotations": {}}
+        kmsplugin->>externalkms: decrypt DEK with remote KEK (same behavior as today)
+        externalkms->>kmsplugin: decrypted DEK
+        kmsplugin->>kubeapiserver: return decrypt response <br/> {"plaintext": "<decrypted DEK>", key_id: "<remote KEK ID>", <br/> "annotations": {}}
+    end
+```
+
+## What’s next?
+For Kubernetes 1.26, we plan to ship another alpha version. As of right now, the alpha API will be ready to be used by KMS plugin authors. We plan to include a reference plugin implementation with the next release, and you'll be able to try out the feature at that time.
+
+You can learn more about KMS v2 by reading the [kubernetes documentation](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/). You can also follow along on the [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/3299-kms-v2-improvements/README.md) to track progress across the coming Kubernetes releases.
+
+## How to get involved
+If you are interested in getting involved in the development of this feature, share feedback, or participate in any other ongoing SIG Auth projects, please reach out on the [#sig-auth](https://kubernetes.slack.com/archives/C0EN96KUY) channel on Kubernetes Slack.
```

Comment:
[sig-auth-kms-dev](https://kubernetes.slack.com/archives/C03035EH4VB) is the dedicated channel for these discussions.